invenio_files_rest/views.py

```diff
@@ -205,56 +262,40 @@ def post(self, **kwargs):
             :statuscode 403: access denied
             :statuscode 500: failed request
         """
-        args = parser.parse(self.post_args, request)
-        try:
-            if args['location_name']:
-                # TODO: Check why query is used directly.
-                location = Location.get_by_name(args['location_name'])
-            else:
-                # Get one of the active locations
-                location = Location.get_default()
-            if not location:
-                abort(400, 'Invalid location.')
-            bucket = Bucket(
-                default_storage_class=current_app.config[
+        with db.session.begin_nested():
+            bucket = Bucket.create(
+                storage_class=current_app.config[
                     'FILES_REST_DEFAULT_STORAGE_CLASS'
                 ],
-                default_location=location.id
+                location=location_name
             )
-            db.session.add(bucket)
-            db.session.commit()
-        except SQLAlchemyError:
-            db.session.rollback()
-            current_app.logger.exception('Failed to create bucket.')
-            abort(500, 'Failed to create bucket.')
-
-        # TODO: Implement serializer
-        return {'json':
-                {'size': bucket.size,
-                 'url': url_for("invenio_files_rest.bucket_api",
-                                bucket_id=bucket.id, _external=True),
-                 'uuid': str(bucket.id)
-                 }
-                }
+        db.session.commit()
+
+        return bucket
 
 
 class BucketResource(ContentNegotiatedMethodView):
     """"Bucket item resource."""
 
+    get_args = {
+        'versions': fields.Boolean(
+            location='query',
+            missing=False,
+        )
+    }
+
     def __init__(self, serializers=None, *args, **kwargs):
         """Constructor."""
         super(BucketResource, self).__init__(
             serializers,
             *args,
             **kwargs
         )
-        self.get_args = {
-            'versions': fields.Boolean(
-                location='query'
-            )
-        }
 
-    def get(self, bucket_id, **kwargs):
+    @use_kwargs(get_args)
+    @pass_bucket
+    @need_bucket_permission('bucket-read')
+    def get(self, versions, bucket=None, **kwargs):
```

Comment:
We need to be able to probably distinguish between allowing list of versions or not.
http://docs.aws.amazon.com/AmazonS3/latest/dev/using-with-s3-actions.html